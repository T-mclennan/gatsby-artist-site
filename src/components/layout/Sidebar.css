.sidebar {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 999;
  display: grid;
  place-items: center;
  opacity: 0;
  transition: var(--transition);
  transform: translateX(-100%);
  background: rgba(255, 255, 255, 0.97);
  box-shadow: 0 8px 32px 0 rgba(31, 38, 135, 0.37);
  backdrop-filter: blur(7.5px);
  -webkit-backdrop-filter: blur(7.5px);
  border: 1px solid rgba(255, 255, 255, 0.18);
}
.show-sidebar {
  opacity: 1;
  transform: translateX(0);
}
.sidebar-links li {
  opacity: 0;
}
.sidebar-links li a {
  display: block;
  text-align: center;
  text-transform: capitalize;
  color: var(--clr-primary-2);
  letter-spacing: var(--spacing);
  margin-bottom: 0.75rem;
  font-size: 2rem;
  transition: var(--transition);
  border-radius: var(--radius);
}
.sidebar-links li a:hover {
  /* background: var(--clr-primary-9); */
  color: var(--clr-primary-5);
}
.close-btn {
  position: absolute;
  right: 4.75%;
  top: 2.75%;
  font-size: 2.5rem;
  background: transparent;
  border-color: transparent;
  color: var(--clr-primary-2);
  cursor: pointer;
}
@media screen and (min-width: 992px) {
  .sidebar {
    transform: translateX(-100%);
  }
}
/*
=============== 
Sidebar Animation
===============
*/
.sidebar-links-animate li {
  animation: slideRight 0.4s ease-in-out 0.1s forwards;
}
.sidebar-links-animate li:nth-of-type(1) {
  animation-delay: 0.08s;
}
.sidebar-links-animate li:nth-of-type(2) {
  animation-delay: 0.16s;
}
.sidebar-links-animate li:nth-of-type(3) {
  animation-delay: 0.24s;
}
.sidebar-links-animate li:nth-of-type(4) {
  animation-delay: 0.32s;
}
.sidebar-links-animate li:nth-of-type(5) {
  animation-delay: 0.4s;
}
@keyframes slideRight {
  0% {
    transform: translateX(-200px);
    opacity: 0;
  }
  100% {
    transform: translateX(0);
    opacity: 1;
  }
}
/* .sidebar-icons li {
  opacity: 0;
  animation: slideUp 0.5s ease-in-out 0.3s forwards;
}
.sidebar-icons li:nth-of-type(1) {
  animation-delay: 0.25s;
}
.sidebar-icons li:nth-of-type(2) {
  animation-delay: 0.5s;
}
.sidebar-icons li:nth-of-type(3) {
  animation-delay: 0.75s;
}
.sidebar-icons li:nth-of-type(4) {
  animation-delay: 1s;
}
.sidebar-icons li:nth-of-type(5) {
  animation-delay: 1.25s;
}
@keyframes slideUp {
  0% {
    transform: translateY(200px);
    opacity: 0;
  }
  100% {
    transform: translateY(0);
    opacity: 1;
  }
} */
