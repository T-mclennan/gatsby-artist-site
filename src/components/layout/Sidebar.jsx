import React from "react";
import SideLinks from "./SideLinks";
import { FaTimes } from "react-icons/fa";
import PropTypes from 'prop-types';
import './Sidebar.css'; 

const Sidebar = ({
  isOpen,
  toggleSidebar}) => {
  return <aside className={`sidebar ${isOpen? 'show-sidebar' : ''}`} data-test="component-sidebar" >
    <button className="close-btn" onClick={toggleSidebar} data-test="component-close-button">
      <FaTimes />
    </button>
    <div className="side-container">
      <SideLinks styleClass={`${isOpen? 'sidebar-links-animate' : ''} sidebar-links`} toggle={toggleSidebar}/>
    </div>

  </aside>
}

Sidebar.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  toggleSidebar: PropTypes.func,
}

export default Sidebar