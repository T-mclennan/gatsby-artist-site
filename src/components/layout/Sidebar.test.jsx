import React from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react"
import Sidebar from "./Sidebar";
import {checkProps} from '../../test/testUtils';


const defaultProps = {
  toggleSidebar: ()=>{}, 
  isOpen: true,
}

describe("Sidebar", () => {
  it("Has properly rendered", () => {
    const tree = renderer
      .create(<Sidebar {...defaultProps}/>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  }),

  it("Shows sidebar when isOpen is true", () => {
    const { getByTestId } = render(<Sidebar toggleSidebar={() => {}} isOpen={true}/>)
    expect(getByTestId("component-sidebar")).toHaveClass('show-sidebar')
  })

  it("Hides sidebar when isOpen is false", () => {
    const { getByTestId } = render(<Sidebar toggleSidebar={() => {}} isOpen={false}/>)
    expect(getByTestId("component-sidebar")).not.toHaveClass('show-sidebar')
  })

  it("Shows appropriate links", () => {
    const { getByTestId } = render(<Sidebar {...defaultProps}/>);
    const parent = getByTestId('component-sidebar');
    const child = getByTestId('side-links');
    expect(parent).toContainElement(child);
  })

  it('Does not throw warning with expected props', () => {
    checkProps(Sidebar, defaultProps);
  })
})
