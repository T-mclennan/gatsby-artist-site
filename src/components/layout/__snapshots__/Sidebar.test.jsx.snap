// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Has properly rendered 1`] = `
<aside
  className="sidebar show-sidebar"
  data-testid="component-sidebar"
>
  <div
    className="side-container"
  >
    <ul
      className="page-links sidebar-links-animate sidebar-links"
      data-testid="side-links"
    >
      <li>
        <a
          href="/about/"
          onClick={[Function]}
        >
          about
        </a>
      </li>
      <li>
        <a
          href="/projects/"
          onClick={[Function]}
        >
          projects
        </a>
      </li>
      <li>
        <a
          href="/gallery/"
          onClick={[Function]}
        >
          gallery
        </a>
      </li>
      <li>
        <a
          href="/music/"
          onClick={[Function]}
        >
          music
        </a>
      </li>
      <li>
        <a
          href="/contact/"
          onClick={[Function]}
        >
          contact
        </a>
      </li>
    </ul>
  </div>
</aside>
`;
